tmp/lanelet2_extension/lib/query.cpp

```diff
@@ -786,6 +786,9 @@ bool query::getClosestLanelet(
     double pose_yaw = tf2::getYaw(search_pose.orientation);
     for (const auto & llt : candidate_lanelets) {
       lanelet::ConstLineString3d segment = getClosestSegment(search_point, llt.centerline());
+      if (segment.empty()) {
+        continue;
+      }
       double segment_angle = std::atan2(
         segment.back().y() - segment.front().y(), segment.back().x() - segment.front().x());
       double angle_diff = std::abs(autoware_utils::normalize_radian(segment_angle - pose_yaw));
```

Comment:
I prefer not to continue when the returned segment is empty because if all candidate_lanelets return empty segment, then there will be no lanelets set to `closest_lanelet_ptr` even though it returns `found == true`

How about something like the following:
```suggestion
      double angle_diff = std::numeric_limits<double>::max();
      if (segment.empty()) {
          angle_diff = M_PI;
      } else {
          double segment_angle = std::atan2(
            segment.back().y() - segment.front().y(), segment.back().x() - segment.front().x());
          angle_diff = std::abs(autoware_utils::normalize_radian(segment_angle - pose_yaw));
      }
```

Reply:
thanks, that is better

Reply:
and refactoring in 5322d1073af33ea4cdd172c016677d21d6c1f18d